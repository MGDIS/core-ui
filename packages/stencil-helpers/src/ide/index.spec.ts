import { describe, expect, test } from 'vitest';
import { webTypesGenerator, vsCodeGenerator } from '.';
import { JsonDocs } from '@stencil/core/internal';
import jsonDocs from '../../json-doc.test.json';

describe('ide', () => {
  describe('webTypesGenerator', () => {
    test('Should generate web types', () => {
      const webTypes = webTypesGenerator('test', '1.0.0', jsonDocs as JsonDocs, 'http://sb.test/?path=/docs/');
      expect(webTypes).toMatchSnapshot();
    });
  });
  describe('vsCodeGenerator', () => {
    test('Should generate VScode custom data', () => {
      const customData = vsCodeGenerator('1.0.0', jsonDocs as JsonDocs, 'http://sb.test/?path=/docs/', 'http://sources.test/');
      expect(customData).toMatchSnapshot();
    });
  });
});
