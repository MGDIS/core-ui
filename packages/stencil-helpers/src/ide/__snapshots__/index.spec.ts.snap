// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ide > vsCodeGenerator > Should generate VScode custom data 1`] = `
{
  "globalAttributes": [],
  "tags": [
    {
      "attributes": [
        {
          "description": "docs

Type: \`string\`",
          "name": "prop",
          "references": [
            {
              "name": "Storybook",
              "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
            },
            {
              "name": "Sources",
              "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
            },
          ],
          "values": undefined,
        },
        {
          "description": "docs

Type: \`string\`",
          "name": "attr",
          "references": [
            {
              "name": "Storybook",
              "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
            },
            {
              "name": "Sources",
              "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
            },
          ],
          "values": [
            {
              "name": "blu",
            },
            {
              "name": "bli",
            },
          ],
        },
        {
          "description": "docs

Type: \`(string | number)[]\`",
          "name": "multionetype",
          "references": [
            {
              "name": "Storybook",
              "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
            },
            {
              "name": "Sources",
              "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
            },
          ],
          "values": undefined,
        },
        {
          "description": "docs

Type: \`number\`",
          "name": "number",
          "references": [
            {
              "name": "Storybook",
              "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
            },
            {
              "name": "Sources",
              "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
            },
          ],
          "values": undefined,
        },
        {
          "description": "docs

Type: \`boolean\`",
          "name": "boolean",
          "references": [
            {
              "name": "Storybook",
              "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
            },
            {
              "name": "Sources",
              "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
            },
          ],
          "values": undefined,
        },
        {
          "description": "docs

Type: \`{ type: string, value: "blu" | "bli" }\`",
          "name": "objecttype",
          "references": [
            {
              "name": "Storybook",
              "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
            },
            {
              "name": "Sources",
              "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
            },
          ],
          "values": undefined,
        },
        {
          "description": "docs

Type: \`"blu" | "bli"\`",
          "name": "multitype",
          "references": [
            {
              "name": "Storybook",
              "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
            },
            {
              "name": "Sources",
              "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
            },
          ],
          "values": undefined,
        },
        {
          "description": "docs

Type: \`number | string\`",
          "name": "multitypewithstring",
          "references": [
            {
              "name": "Storybook",
              "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
            },
            {
              "name": "Sources",
              "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
            },
          ],
          "values": undefined,
        },
        {
          "description": "docs

Type: \`number[] | string[]\`",
          "name": "arraytypes",
          "references": [
            {
              "name": "Storybook",
              "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
            },
            {
              "name": "Sources",
              "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
            },
          ],
          "values": undefined,
        },
      ],
      "description": "Attributes:
- \`prop\`: docs
- \`multionetype\`: docs
- \`number\`: docs
- \`boolean\`: docs
- \`objecttype\`: docs
- \`multitype\`: docs
- \`multitypewithstring\`: docs
- \`arraytypes\`: docs

Properties:
- \`attr\`: docs

Methods:
- \`method\`: docs

Events:
- \`event\`: docs

Listeners:
- \`keydown\`

Slots:
- \`\`: 
- \`title\`: 

",
      "name": "my-comp",
      "references": [
        {
          "name": "Storybook",
          "url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
        },
        {
          "name": "Sources",
          "url": "http://sources.test/src/components/atoms/my-comp/my-comp.tsx",
        },
      ],
    },
    {
      "attributes": [],
      "description": "overview

",
      "name": "my-comp",
      "references": [
        {
          "name": "Storybook",
          "url": undefined,
        },
        {
          "name": "Sources",
          "url": undefined,
        },
      ],
    },
  ],
  "valueSets": [],
  "version": "1.0.0",
}
`;

exports[`ide > webTypesGenerator > Should generate web types 1`] = `
{
  "$schema": "https://json.schemastore.org/web-types",
  "contributions": {
    "html": {
      "elements": [
        {
          "attributes": [
            {
              "description": "docs

Type: \`string\`",
              "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
              "name": "prop",
              "value": {
                "default": undefined,
                "required": true,
                "type": "string",
              },
            },
            {
              "description": "docs

Type: \`(string | number)[]\`",
              "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
              "name": "multionetype",
              "value": {
                "default": undefined,
                "required": true,
                "type": "(string | number)[]",
              },
            },
            {
              "description": "docs

Type: \`number\`",
              "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
              "name": "number",
              "value": {
                "default": undefined,
                "required": true,
                "type": "number",
              },
            },
            {
              "description": "docs

Type: \`boolean\`",
              "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
              "name": "boolean",
              "value": {
                "default": undefined,
                "required": true,
                "type": "boolean",
              },
            },
            {
              "description": "docs

Type: \`{ type: string, value: "blu" | "bli" }\`",
              "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
              "name": "objecttype",
              "value": {
                "default": undefined,
                "required": true,
                "type": "{ type: string, value: "blu" | "bli" }",
              },
            },
            {
              "description": "docs

Type: \`"blu" | "bli"\`",
              "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
              "name": "multitype",
              "value": {
                "default": undefined,
                "required": true,
                "type": ""blu" | "bli"",
              },
            },
            {
              "description": "docs

Type: \`number | string\`",
              "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
              "name": "multitypewithstring",
              "value": {
                "default": undefined,
                "required": true,
                "type": "number | string",
              },
            },
            {
              "description": "docs

Type: \`number[] | string[]\`",
              "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
              "name": "arraytypes",
              "value": {
                "default": undefined,
                "required": true,
                "type": "number[] | string[]",
              },
            },
          ],
          "description": "Attributes:
- \`prop\`: docs
- \`multionetype\`: docs
- \`number\`: docs
- \`boolean\`: docs
- \`objecttype\`: docs
- \`multitype\`: docs
- \`multitypewithstring\`: docs
- \`arraytypes\`: docs

Properties:
- \`attr\`: docs

Methods:
- \`method\`: docs

Events:
- \`event\`: docs

Listeners:
- \`keydown\`

Slots:
- \`\`: 
- \`title\`: 

",
          "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
          "js": {
            "events": [
              {
                "description": "docs",
                "name": "event",
              },
            ],
            "properties": [
              {
                "description": "docs

Type: \`string\`",
                "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
                "name": "prop",
                "value": {
                  "default": undefined,
                  "required": true,
                  "type": "string",
                },
              },
              {
                "description": "docs

Type: \`string\`",
                "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
                "name": "attr",
                "value": {
                  "default": undefined,
                  "required": true,
                  "type": "string",
                },
              },
              {
                "description": "docs

Type: \`(string | number)[]\`",
                "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
                "name": "multionetype",
                "value": {
                  "default": undefined,
                  "required": true,
                  "type": "(string | number)[]",
                },
              },
              {
                "description": "docs

Type: \`number\`",
                "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
                "name": "number",
                "value": {
                  "default": undefined,
                  "required": true,
                  "type": "number",
                },
              },
              {
                "description": "docs

Type: \`boolean\`",
                "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
                "name": "boolean",
                "value": {
                  "default": undefined,
                  "required": true,
                  "type": "boolean",
                },
              },
              {
                "description": "docs

Type: \`{ type: string, value: "blu" | "bli" }\`",
                "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
                "name": "objecttype",
                "value": {
                  "default": undefined,
                  "required": true,
                  "type": "{ type: string, value: "blu" | "bli" }",
                },
              },
              {
                "description": "docs

Type: \`"blu" | "bli"\`",
                "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
                "name": "multitype",
                "value": {
                  "default": undefined,
                  "required": true,
                  "type": ""blu" | "bli"",
                },
              },
              {
                "description": "docs

Type: \`number | string\`",
                "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
                "name": "multitypewithstring",
                "value": {
                  "default": undefined,
                  "required": true,
                  "type": "number | string",
                },
              },
              {
                "description": "docs

Type: \`number[] | string[]\`",
                "doc-url": "http://sb.test/?path=/docs/atoms-my-comp--docs",
                "name": "arraytypes",
                "value": {
                  "default": undefined,
                  "required": true,
                  "type": "number[] | string[]",
                },
              },
            ],
          },
          "name": "my-comp",
        },
        {
          "attributes": [],
          "description": "overview

",
          "doc-url": undefined,
          "js": {
            "events": [],
            "properties": [],
          },
          "name": "my-comp",
        },
      ],
    },
  },
  "description-markup": "markdown",
  "name": "test",
  "version": "1.0.0",
}
`;
